library/std/src/os/uefi/ffi.rs

```diff
@@ -0,0 +1,73 @@
+use crate::ffi::{OsStr, OsString};
+use crate::sealed::Sealed;
+use crate::sys_common::ucs2;
+
+#[unstable(feature = "uefi_std", issue = "100499")]
+pub trait OsStrExt: Sealed {
+    /// This function does not do any allocation
+    fn to_ucs2<'a>(&'a self) -> ucs2::EncodeUcs2<'a>;
+
+    /// Creates a UCS-2 Vec which can be passed for FFI.
+    /// Note: This function will replace `NULL` and other characters which are not valid in UEFI
+    /// strings with `std::sys_common::ucs::Ucs2Char::REPLACEMENT_CHARACTER`
+    fn to_ffi_string(&self) -> Vec<u16> {
+        let mut v: Vec<u16> = self
+            .to_ucs2()
+            .map(|x| match x {
+                Ok(c) => c,
+                Err(_) => ucs2::Ucs2Char::REPLACEMENT_CHARACTER,
+            })
+            .map(u16::from)
+            .collect();
+        v.push(0);
+        v.shrink_to_fit();
```

Comment:
This is probably not needed.

Reply:
By this, do you mean line 23?

Reply:
Well, this has been changed with using UTF-16.